web/modules/custom/cyber_breadcrumbs/src/BasicBreadcrumb.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Drupal\cyber_breadcrumbs;
+
+/**
+ * @file
+ * Generates the breadcrumb trail for content including:
+ * - Publication
+ *
+ * In the format:
+ * > Home
+ * > current-page-title
+ *
+ * > <front>
+ * > /current-page-title
+ */
+use Drupal\Core\Breadcrumb\Breadcrumb;
+use Drupal\Core\Breadcrumb\BreadcrumbBuilderInterface;
+use Drupal\Core\Controller\TitleResolverInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Url;
+use Drupal\node\NodeInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class BasicBreadcrumb implements BreadcrumbBuilderInterface {
+
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+  /**
+   * Node object, or null if on a non-node page.
+   *
+   * @var \Drupal\node\Entity\Node
+   */
+  protected $node;
+  /**
+   * The title resolver.
+   *
+   * @var \Drupal\Core\Controller\TitleResolverInterface
+   */
+  protected $titleResolver;
+
+  /**
+   * Class constructor.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, TitleResolverInterface $title_resolver) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->titleResolver = $title_resolver;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('title_resolver')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applies(RouteMatchInterface $route_match) {
+    $match = FALSE;
+    $route_name = $route_match->getRouteName();
+    if ($route_name == 'entity.node.canonical') {
+      $this->node = $route_match->getParameter('node');
+      if ($this->node instanceof NodeInterface == FALSE) {
+        $this->node = $this->entityTypeManager->getStorage('node')->load($this->node);
+      }
+      if (!empty($this->node)) {
+        $match = $this->node->bundle() == 'basic_page';
+      }
+    }
+    return $match;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(RouteMatchInterface $route_match) {
+    $breadcrumb = new Breadcrumb();
+    $title_resolver = $this->titleResolver->getTitle(\Drupal::request(), $route_match->getRouteObject());
```

Comment:
You should be able to get the request from the service container too :)

Try replacing `\Drupal::request()` with `CurrentRouteMatch` (`current_route_match` service). See NIDirect custom code repo under: `nidirect_contacts/src/Plugin/Block/ContactAzBlock.php` for an example.

Reply:
Whoops, scratch that. You want the request stack service (`request_stack`) which is a Symfony component rather than a Drupal one. Example at `nidirect_contacts/src/Controller/ContactListingController.php`